dev/io.openliberty.jakartaee10.internal_fat/fat/src/io/openliberty/jakartaee10/internal/tests/FeatureUtilities.java

```diff
@@ -0,0 +1,166 @@
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.rules.repeater.EE7FeatureReplacementAction;
+import componenttest.rules.repeater.EE8FeatureReplacementAction;
+import componenttest.rules.repeater.JakartaEE10Action;
+import componenttest.rules.repeater.JakartaEE9Action;
+import componenttest.rules.repeater.MicroProfileActions;
+import componenttest.rules.repeater.RepeatActions.EEVersion;
+
+/**
+ *
+ */
+public class FeatureUtilities {
+
+    private static final Class<?> c = FeatureUtilities.class;
+
+    /**
+     * Returns the set of all public Java/Jakarta EE feature short names
+     *
+     * @return the set of short names
+     */
+    public static Set<String> allEeFeatures() {
+        Set<String> features = new HashSet<>();
+        features.addAll(EE7FeatureReplacementAction.EE7_FEATURE_SET);
+        features.addAll(EE8FeatureReplacementAction.EE8_FEATURE_SET);
+        features.addAll(JakartaEE9Action.EE9_FEATURE_SET);
+        features.addAll(JakartaEE10Action.EE10_FEATURE_SET);
+
+        // EE-related features which aren't in one of the feature sets
+        features.add("appSecurity-1.0");
+        features.add("jsp-2.2");
+        features.add("websocket-1.0");
+
+        return features;
+    }
+
+    /**
+     * Returns the set of all public MicroProfile feature short names
+     *
+     * @return the set of short names
+     */
+    public static Set<String> allMpFeatures() {
+        return Stream.concat(MicroProfileActions.ALL.stream(),
+                             MicroProfileActions.STANDALONE_ALL.stream())
+                        .flatMap(s -> s.getFeatures().stream())
+                        .collect(Collectors.toSet());
+    }
+
+    /**
+     * Returns the set of public MicroProfile features which are compatible with a given Java/Jakarta EE version
+     *
+     * @param eeVersion the EE version
+     * @return the set of compatible MP feature short names
+     */
+    public static Set<String> compatibleMpFeatures(EEVersion eeVersion) {
+        return Stream.concat(MicroProfileActions.ALL.stream(),
+                             MicroProfileActions.STANDALONE_ALL.stream())
+                        .filter(s -> s.getEEVersion() == eeVersion)
+                        .flatMap(s -> s.getFeatures().stream())
+                        .collect(Collectors.toSet());
+    }
+
+    /**
+     * Returns the set of public test features
+     *
+     * @return the set of short names of public test features
+     */
+    public static Set<String> allTestFeatures() {
+        Set<String> testFeatures = new HashSet<>();
+        testFeatures.add("componenttest-1.0");
+        testFeatures.add("componenttest-2.0");
+        testFeatures.add("txtest-1.0");
+        testFeatures.add("txtest-2.0");
+        testFeatures.add("ejbTest-1.0");
+        testFeatures.add("ejbTest-2.0");
+        testFeatures.add("enterpriseBeansTest-2.0");
```

Comment:
Not sure why you added enterpriseBeansTest-2.0.  It is marked IBM-Test-Feature, so it wouldn't be in the list already.

Reply:
oh I see why